### PR TITLE
Added Haxe standard library Socket class support for the fast_select fun...

### PR DESCRIPTION
**Pull request by bjitivo**
...ction,

which operates just like the select function but does not churn nearly as many
objects by modifing the input arrays rather than creating new ones.

### CODE REVIEW

File: extra/haxelib_src [hunk 1]
```diff
@@ -1 +1 @@
-Subproject commit 68f25d97dc991516c02a2665081ba5cc0d2ae23e
+Subproject commit 0dfb33242a28d5a89178c1e48968174646e0fefe
```
Comment (andyli) [inline extra/haxelib_src]: Not sure how git makes it so easy to mess up submodule reference... :P

Comment (bjitivo) [inline extra/haxelib_src]: Sorry, git is confusing enough sometimes without subprojects.  Are you saying that the commit is flawed somehow?

**Comment by bjitivo** [comment]
I re-did the commit, without the change to the submodule thing, I hope this is OK.

**Comment by andyli** [comment]
I can see that you've removed the submodule change, thanks!

**Comment by ncannasse** [comment]
There seems to be pending epoll PR, maybe we should integrate epoll instead of optimizing select ? 
https://github.com/HaxeFoundation/neko/pull/67

**Comment by bjitivo** [comment]
One problem I have with that epoll implementation is that it is churny.  Allocating three arrays every poll is alot of churn.

Also, epoll is a Linux specific system call.  It will not be portable to other platforms (as the implementation shows), and so provides only limited benefit.

Our primary interest in creating fast_select was avoiding the churn, on all platforms (including Java), which the epoll pull request does not help with.

**Comment by Simn** [comment]
This PR no longer merges. Sorry for not looking into it earlier, but at this point the best we can do is close it.

**Pull request closed without merging**